diff -ru glib-2.12.13/glib/gconvert.c glib-2.12.13+iPhone/glib/gconvert.c
--- glib-2.12.13/glib/gconvert.c	2007-07-16 18:57:12.000000000 +0000
+++ glib-2.12.13+iPhone/glib/gconvert.c	2008-06-18 04:14:55.000000000 +0000
@@ -41,13 +41,6 @@
 
 #include "glibintl.h"
 
-#if defined(USE_LIBICONV_GNU) && !defined (_LIBICONV_H)
-#error GNU libiconv in use but included iconv.h not from libiconv
-#endif
-#if !defined(USE_LIBICONV_GNU) && defined (_LIBICONV_H)
-#error GNU libiconv not in use but included iconv.h is from libiconv
-#endif
-
 #include "galias.h"
 
 GQuark 

